src/Adapter/Product/Repository/ProductRepository.php

```diff
@@ -412,6 +414,20 @@ public function searchCombinations(
         return $qb->execute()->fetchAllAssociative();
     }
 
+    public function getProductTaxRulesGroupId(ProductId $productId): TaxRulesGroupId
+    {
+        $result = $this->connection->createQueryBuilder()
+            ->addSelect('p.id_tax_rules_group')
+            ->from($this->dbPrefix . 'product', 'p')
+            ->where('id_product = :productId')
+            ->setParameter('productId', $productId->getValue())
+            ->execute()
+            ->fetchOne()
+        ;
+
+        return new TaxRulesGroupId((int) $result);
```

Comment:
Mhm I see. This change came from other PR after rebase. I will just move the method up now (cuz its public and should be above private methods). But Im afraid modifying it as It is out of scope.
@FabienPapet I see its your change, I wonder must product always have a taxRulesGroupId? It might be a mistake in a contract, could you check it and fix in a dedicated PR?

Reply:
Ok if it's considered as a valid value

Reply:
We should never handle `0` values (and sorry about that this is the case here) .If the TaxGroupRuleId does not exists, I think it's better to throw a `RuntimeException` if the `TaxGroupRuleId` should always exist, otherwise a proper exception would be better appropriate.

Reply:
Just to make it clear, Im not fixing it in this PR, because its absolutely out of scope, so as it was your change, @FabienPapet (at least I see that by history), I trust it to you to fix in a dedicated PR :stuck_out_tongue: 
And yes, if TaxRulesGroupId is optional value it should insipre from similar cases like https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Domain/Manufacturer/ValueObject/NoManufacturerId.php.
Else if its not optional, then this method should throw TaxRulesGroupIdNotFoundException.

Reply:
I agree in other use cases we used a common interface a `NoValueObject` dedicated classes for the 0 value But it's not in this PR's scope and can be improved in another PR (it will also create BC since `TaxRulesGroupId` will no longer accept the 0 value)

The `getProductTaxRulesGroupId` should not trigger an `TaxRulesGroupIdNotFoundException` though since the 0 value is acceptable, it should return a VO matching the interface that could be either `TaxRulesGroupId` or `NoTaxRulesGroupId`

Reply:
I'm ok but plz let's think about fixing it soon.